### DISCUSSION TITLE
Application Exiting Randomly

### DISCUSSION THREAD

**Discussion by viperdudeuk** [abraunegg/onedrive#2465] [General]
I have just installed the Onedrive for Linux app. I have authenticated against my OneDrive account ok

However when I call

onedrive --synchronize

The app starts downloading the data but after a few minutes the program exists. I have around 56000 items on my OneDrive. Have to restart we very few minutes is going to be very painful.

Anyone know what I am doing wrong here?

**Comment by abraunegg** [comment]
@viperdudeuk 
> I have just installed the Onedrive for Linux app

What application version are you using - what is the output of `onedrive --version` ? It is always more helpful to include this sort of data point to quickly identify if you have run into a known issue. Additionally what Linux distribution are you using?

> onedrive --synchronize
>
>The app starts downloading the data but after a few minutes the program exists. I have around 56000 items on my OneDrive. Have to >restart we very few minutes is going to be very painful.

A self help item here is to increase your application output to understand why this is occurring - `onedrive --synchronize --verbose` or even enable debug logging - `onedrive --synchronize --verbose --verbose`

To me this potentially sounds like a mis-configured firewall that is dropping your active connection, that the firewall thinks is not active.

Additionally, there are several known issues depending on what application version you are using. Please read: https://github.com/abraunegg/onedrive/blob/master/docs/known-issues.md#application-stops-running-without-any-visible-reason

**Comment by viperdudeuk** [comment]
Well I installed OneDrive as a service and since then has worked perfectly. Synced everything down from OneDrive and I tested a local created file and it was immediately upload. 

Not sure what the issue was but running as a service is the way forward.

Thank you for all your efforts in making this app.

**Comment by abraunegg** [comment]
@viperdudeuk  

> Well I installed OneDrive as a service and since then has worked perfectly. Synced everything down from OneDrive and I tested a local created file and it was immediately upload.
> 
> Not sure what the issue was but running as a service is the way forward.

The systemd service will be masking your underlying issue by restarting the client when it exits - so you potentially still have a problem that you need to fix.

As you have missed responding to my questions, please ensure you are running the correct application version (currently v2.4.25) to ensure you have a client that is stable and relatively bug free.
 
> Thank you for all your efforts in making this app.

Thanks for the feedback.

**Reply by viperdudeuk** [reply]
I am away now for over a week but i will check the version when I return.
